internal/controller/controllers/local_cluster_import_controller.go

```diff
@@ -0,0 +1,517 @@
+	if err != nil {
+		return errors.Wrap(err, "could not create or AgentClusterInstall")
+	}
+	return nil
+}
+
+func (r *LocalClusterImportReconciler) createOrUpdateClusterDeployment(ctx context.Context, pullSecret *v1.Secret, dns *configv1.DNS, kubeConfigSecret *v1.Secret, instance *aiv1beta1.AgentServiceConfig) error {
+	if pullSecret == nil {
+		return errors.New("pull-secret is not defined, unable to perform local-cluster import")
+	}
+	if dns == nil {
+		return errors.New("cluster dns is not defined, unable to perform local-cluster import")
+	}
+	if kubeConfigSecret == nil {
+		return errors.New("kubeconfig secret is not defined, unable to perform local-cluster import")
+	}
+	if instance == nil {
+		return errors.New("agentServiceConfig is not defined, unable to perform local-cluster import")
+	}
+	clusterDeployment := hivev1.ClusterDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.localClusterName,
+			Namespace: r.localClusterName,
+		},
+	}
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, &clusterDeployment, r.client.Scheme()); err != nil {
+			return err
+		}
+		clusterDeployment.Spec.Installed = true
+		clusterDeployment.Spec.ClusterMetadata = &hivev1.ClusterMetadata{
+			ClusterID:                "",
+			InfraID:                  "",
+			AdminKubeconfigSecretRef: v1.LocalObjectReference{Name: fmt.Sprintf("%s-admin-kubeconfig", r.localClusterName)},
+		}
+		clusterDeployment.Spec.ClusterInstallRef = &hivev1.ClusterInstallLocalReference{
+			Name:    r.localClusterName,
+			Group:   "extensions.hive.openshift.io",
+			Kind:    "AgentClusterInstall",
+			Version: "v1beta1",
+		}
+		clusterDeployment.Spec.Platform = hivev1.Platform{
+			AgentBareMetal: &agent.BareMetalPlatform{
+				AgentSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{"infraenv": "local-cluster"},
+				},
+			},
+		}
+		clusterDeployment.Spec.PullSecretRef = &v1.LocalObjectReference{
+			Name: pullSecret.Name,
+		}
+		clusterDeployment.Spec.ClusterName = r.localClusterName
+		clusterDeployment.Spec.BaseDomain = dns.Spec.BaseDomain
+		return nil
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, r.client, &clusterDeployment, mutateFn)
+	if err != nil {
+		return errors.Wrap(err, "could not create or ClusterDeployment")
+	}
+	return nil
+}
+
+func (r *LocalClusterImportReconciler) importLocalCluster(ctx context.Context, instance *aiv1beta1.AgentServiceConfig) error {
+
+	clusterVersion := &configv1.ClusterVersion{}
+	namespacedName := types.NamespacedName{
+		Name: "version",
+	}
+	err := r.client.Get(ctx, namespacedName, clusterVersion)
+	if err != nil {
+		return errors.Wrap(err, "unable to find cluster version")
+	}
+
+	kubeConfigSecret, err := r.getSecret(ctx, "openshift-kube-apiserver", "node-kubeconfigs")
+	if err != nil {
+		return errors.Wrap(err, "unable to fetch local cluster kubeconfigs")
+	}
+
+	pullSecret, err := r.getSecret(ctx, "openshift-config", "pull-secret")
```

Comment:
These strings should probably be constants so you can share with the predicates.

Reply:
(Same for the kubeconfig)